src/main/java/org/openstreetmap/atlas/generator/sharding/PbfVerifier.java

```diff
@@ -0,0 +1,168 @@
+import org.openstreetmap.atlas.utilities.collections.StringList;
+import org.openstreetmap.atlas.utilities.maps.MultiMap;
+import org.openstreetmap.atlas.utilities.runtime.Command;
+import org.openstreetmap.atlas.utilities.runtime.CommandMap;
+import org.openstreetmap.atlas.utilities.time.Time;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Command that verifies all expected PBF files exist and can be converted to atlas.
+ *
+ * @author jamesgage
+ */
+public class PbfVerifier extends Command
+{
+    private static final Switch<File> SLIPPY_TILE_FILE = new Switch<>("slippyTileFile",
+            "The slippy tile file that lists all pbf files that should have been generated.",
+            File::new, Optionality.REQUIRED);
+    private static final Switch<String> PBF_PATH = new Switch<>("pbfPath",
+            "The path to the pbf root directory", value -> value, Optionality.REQUIRED);
+    private static final int LOGGING_RATE = 1000;
+    private static final Logger logger = LoggerFactory.getLogger(PbfVerifier.class);
+
+    public static MultiPolygon forPolygon(final Polygon polygon)
+    {
+        final MultiMap<Polygon, Polygon> multiMap = new MultiMap<>();
+        multiMap.put(polygon, new ArrayList<>());
+        return new MultiPolygon(multiMap);
+    }
+
+    public static void main(final String[] args)
+    {
+        new PbfVerifier().run(args);
+    }
+
+    public static HashMap<String, Rectangle> parseSlippyTileFile(final Resource slippyTileFile)
+    {
+        final Integer size = Integer.parseInt(slippyTileFile.firstLine());
+        final HashMap<String, Rectangle> shardToBounds = new HashMap<>(size);
+        for (final String line : slippyTileFile.lines())
+        {
+            final StringList splitLine = StringList.split(line, "|");
+            // the form of the slippyTile file is "pbfName.pbf|POLYGON( BOUNDS )"
+            if (splitLine.size() == 2)
+            {
+                final String boundsString = splitLine.get(0);
+                final String name = splitLine.get(1);
+                final Rectangle bounds = Rectangle.forString(boundsString);
+                shardToBounds.put(name, bounds);
+            }
+        }
+        return shardToBounds;
+    }
+
+    public int buildAllPbfs(final List<Resource> pbfFiles)
+    {
+        // try to load each pbf, any that fail should be logged
+        final AtomicInteger count = new AtomicInteger();
+        pbfFiles.parallelStream().forEach(pbfFile ->
+        {
+            final RawAtlasGenerator rawAtlasGenerator = new RawAtlasGenerator(pbfFile,
+                    AtlasLoadingOption.createOptionWithNoSlicing(), MultiPolygon.MAXIMUM);
+            try
+            {
+                rawAtlasGenerator.buildNoTrim();
+                final int currentCount = count.getAndIncrement();
+                if (currentCount % LOGGING_RATE == 0)
+                {
+                    logger.info("Processed " + currentCount + " PBF files.");
```

Comment:
Nit: probably cleaner to do `logger.info("Processed {} PBF files.", currentCount);`

Reply:
For whatever reason I find the former more readable, but I think the rest of the team agrees with you! I'll switch them over :)